integrations/tests/kube_cri_apis/main.go

```diff
@@ -16,22 +16,38 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 
 	"github.com/postmanlabs/postman-insights-agent/integrations/cri_apis"
 	"github.com/postmanlabs/postman-insights-agent/integrations/kube_apis"
 	"github.com/postmanlabs/postman-insights-agent/printer"
+	coreV1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
-func k8s_funcs() (string, error) {
-	// Initialize KubeClient
-	kubeClient, err := kube_apis.NewKubeClient()
-	if err != nil {
-		return "", fmt.Errorf("failed to create KubeClient: %v", err)
+func k8s_watcher(kubeClient kube_apis.KubeClient) {
+	// Watch for pod events
+	for event := range kubeClient.EventWatch.ResultChan() {
+		printer.Infof("Received event: %v\n", event.Type)
+		switch event.Type {
+		case watch.Added, watch.Deleted:
+			if e, ok := event.Object.(*coreV1.Event); ok {
+				jsonData, err := json.Marshal(e)
+				if err != nil {
+					printer.Errorf("Failed to marshal event data: %v\n", err)
+					continue
+				}
+				printer.Infof("Event data: %s\n", string(jsonData))
+			}
+		default:
+			printer.Infof("Unhandled event type: %v\n", event.Type)
```

Comment:
This seems like it could be very high-rate, should it be Debug?  Or throttled in some other way?  Or eliminate common cases that we know occur?